src/app/features/approver/approver.context.ts

```diff
@@ -0,0 +1,35 @@
+import { createContext, useContext } from 'react';
+
+import { useOnMount } from '@app/common/hooks/use-on-mount';
+import { type ChildRegister, useRegisterChildren } from '@app/common/hooks/use-register-children';
+
+type ApproverChildren = 'header' | 'actions' | 'advanced' | 'section' | 'subheader';
+
+interface ApproverContext extends ChildRegister<ApproverChildren> {
+  isDisplayingAdvancedView: boolean;
+  setIsDisplayingAdvancedView(val: boolean): void;
+}
+
+const approverContext = createContext<ApproverContext | null>(null);
+
+export const ApproverProvider = approverContext.Provider;
+
+export function useApproverContext() {
+  const context = useContext(approverContext);
+  if (!context) throw new Error('`useApproverContext` must be used within a `ApproverProvider`');
```

Comment:
Are backticks needed here?

Reply:
I don't think so but it is nice to read this way 

![Screenshot 2024-04-12 at 06 17 00](https://github.com/leather-wallet/extension/assets/2938440/b9f4491a-7f2f-4dd8-a6ef-d6312ab9c960)